File: 프로그래머스/lv0/120838. 모스부호 （1）/모스부호 （1）.cs
using System;
using System.Linq;
using System.Collections;
using System.Text;

public class Solution {
    
    readonly string[] morse = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
    
    public string solution(string letter) {
        
        string answer = "";
        
        string[] splitedStrArray = letter.Split(' ');
        var temp = splitedStrArray.Where(n => true).Select(n => Array.IndexOf(morse, n) + 97)
                                       .Select(n => Convert.ToChar(n));
        
        StringBuilder sb = new StringBuilder();
        foreach(char item in temp)
            sb.Append(item);
        
        return sb.ToString();
    }
}

File: 프로그래머스/0/120838. 모스부호 （1）/모스부호 （1）.cpp
#include <string>
#include <vector>
#include <map>
#include <sstream>

using namespace std;

string solution(string letter) {
    
    map<string, string> morse = { {".-","a"},{"-...","b"},{"-.-.","c"},{"-..","d"},{".","e"},{"..-.","f"},
                {"--.","g"},{"....","h"},{"..","i"},{".---","j"},{"-.-","k"},{".-..","l"},
                {"--","m"},{"-.","n"},{"---","o"},{".--.","p"},{"--.-","q"},{".-.","r"},
                {"...","s"},{"-","t"},{"..-","u"},{"...-","v"},{".--","w"},{"-..-","x"},
                {"-.--","y"},{"--..","z"}
                };
    
    string answer;
    stringstream ss(letter);
    string temp;
    
    while (ss >> temp)
    {
        answer += morse[temp];
    } 
    
    return answer;
}